### DISCUSSION TITLE
make an elif block

### DISCUSSION THREAD

**Discussion by theTWIXhunter** [ScratchAddons/ScratchAddons#7002] [Ideas] [answered]
Is it possible to make an addon that makes an if block in an else block look like an elif leg to the first 'if then else'. For non-scratchaddons-users, this could just look like normal

**Comment by BroJac5246** [comment, marked as answer]
I don't think this is rejected if someone wants to make it but the conclusion last time this came up was that it's very hard to display one block while having something else in the backend.

I'm with you, though! Elifs would be super useful!

**Reply by WorldLanguages** [reply]
Yes, there are multiple other issues/discussions that talk about showing a block differently than the internal representation of the block - but we haven't seen a demo of that yet. I think @pufferfish101007 was talking a look recently, but I might be wrong.

**Reply by pufferfish101007** [reply]
Indeed. See https://github.com/pufferfish101007/scratchaddons/tree/custom-reporters for a (bad, and to be improved) example of how this could be done. It is fairly trivialish to create a 'new' block and transform it to/from vanilla scratch blocks on save/load. The way that branch does it is bad, it should not have to be trapping the vm before the project loads; if anyone's interested in trying this out I'd recommend lazily transforming `vm.editingTarget.blocks` each time the editing target's changed. I could give this a go some time (I'd probably be able to create a small PoC soon).
